[🐞] 1.4.1 breaks node/express deployment due to wrong import paths of Qwik-city modules in entry.*.mjs

### Which component is affected?

Qwik Rollup / Vite plugin

### Describe the bug

I am using node/express for deployment.

in 1.4.0, a build generates an entry.express.mjs that imports these modules from the same directory as the entry file is located:

```
entry.express.mjs@1.4.0:

import { getNotFound } from "./@qwik-city-not-found-paths.js";
import { isStaticPath } from "./@qwik-city-static-paths.js"; 
```

In 1.4.1, the import path's are now absolute, referring to a location that only exists during the docker file build stage:

```
entry.express.mjs@1.4.1:

import { getNotFound } from "/build/packages/qwik/server/@qwik-city-not-found-paths.js";
import { isStaticPath } from "/build/packages/qwik/server/@qwik-city-static-paths.js"; 
```

This causes the project to not being able to be started with `node ./server/entry.express.mjs` which is how it was successfully working in 1.4.0 because the directory does not exist.

Obviously this doesn't affect local build/run since absolute paths on the same filesystem are working just fine, but with an intermediary build stage, it fails.

It also seems to affect other deployment methods. (see discord)


The error message in production:

```
2024-01-26T14:13:08.978360686Z  2024-01-26T14:13:08.978362670Z Node.js v20.11.0 2024-01-26T14:13:14.676704118Z node:internal/modules/esm/resolve:264 2024-01-26T14:13:14.676762088Z throw new ERR_MODULE_NOT_FOUND( 2024-01-26T14:13:14.676768090Z ^ 2024-01-26T14:13:14.676771397Z  2024-01-26T14:13:14.676775164Z Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/build/packages/qwik/server/@qwik-city-not-found-paths.js' imported from /app/server/entry.express.mjs 2024-01-26T14:13:14.676778580Z at finalizeResolution (node:internal/modules/esm/resolve:264:11) 2024-01-26T14:13:14.676781896Z at moduleResolve (node:internal/modules/esm/resolve:917:10) 2024-01-26T14:13:14.676785333Z at defaultResolve (node:internal/modules/esm/resolve:1130:11) 2024-01-26T14:13:14.676789031Z at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:396:12) 2024-01-26T14:13:14.676792798Z at ModuleLoader.resolve (node:internal/modules/esm/loader:365:25) 2024-01-26T14:13:14.676796255Z at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:240:38) 2024-01-26T14:13:14.676801865Z at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:85:39) 2024-01-26T14:13:14.676810182Z at link (node:internal/modules/esm/module_job:84:36) { 2024-01-26T14:13:14.676813938Z code: 'ERR_MODULE_NOT_FOUND', 2024-01-26T14:13:14.676817545Z url: 'file:///build/packages/qwik/server/@qwik-city-not-found-paths.js' 2024-01-26T14:13:14.676821172Z } 2024-01-26T14:13:14.676824658Z  2024-01-26T14:13:14.676828065Z Node.js v20.11.0 
```


I cannot provide a reproduction as it is a runtime error in a docker image after build.

### Reproduction

none

### Steps to reproduce

_No response_

### System Info

```shell
System:
    OS: macOS 14.2.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 13.64 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.0 - ~/.nvm/versions/node/v20.11.0/bin/node
    Yarn: 4.0.2 - ~/.nvm/versions/node/v20.11.0/bin/yarn
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.0/bin/npm
  Browsers:
    Brave Browser: 121.1.62.153
    Chrome: 121.0.6167.85
    Safari: 17.2.1
```


### Additional Information

_No response_

--- a/packages/qwik-city/buildtime/vite/plugin.ts
+++ b/packages/qwik-city/buildtime/vite/plugin.ts
@@ -137,13 +137,13 @@ function qwikCityPlugin(userOpts?: QwikCityVitePluginOptions): any {
       }
       if (id === STATIC_PATHS_ID) {
         return {
-          id: join(outDir!, RESOLVED_STATIC_PATHS_ID),
+          id: './' + RESOLVED_STATIC_PATHS_ID,
           external: true,
         };
       }
       if (id === NOT_FOUND_PATHS_ID) {
         return {
-          id: join(outDir!, RESOLVED_NOT_FOUND_PATHS_ID),
+          id: './' + RESOLVED_NOT_FOUND_PATHS_ID,
           external: true,
         };
       }

--- a/packages/qwik/src/optimizer/src/plugins/rollup.ts
+++ b/packages/qwik/src/optimizer/src/plugins/rollup.ts
@@ -195,6 +195,15 @@ export function normalizeRollupOutputOptions(
         outputOpts.chunkFileNames = 'build/[name].js';
       }
     }
+  } else if (opts.buildMode === 'production') {
+    // server production output
+    // everything in same dir so './@qwik-city...' imports work from entry and chunks
+    if (!outputOpts.chunkFileNames) {
+      outputOpts.chunkFileNames = 'q-[hash].js';
+    }
+    if (!outputOpts.assetFileNames) {
+      outputOpts.assetFileNames = 'assets/[hash].[ext]';
+    }
   }
 
   if (opts.target === 'client') {

